### PR TITLE
Unit test

### PR DESCRIPTION
**Pull request by yuvenalmash**
## Changes Summary
- Add unit tests for all the methods in all the classes:
    - Book
    - Person
    - Student
    - Teacher
    - Classroom
    - Rental
    - And all classes created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  it 'should return the author of the book' do
+    expect(@book.author).to eql('J.R.R. Tolkien')
+  end
+
+  it 'check the be_instance_of book' do
+    expect(@book).to be_instance_of(Book)
+  end
+end
```
Comment (tresorsawasawa) [inline spec/book_spec.rb]: - [x] As required for the project to have unit tests for all the `methods` in all the `classes`, Kindly ensure that you add tests on the `add_rental` method in the `Book` class

File: spec/capitalize_decorator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require_relative '../capitalize_decorator'
+
+describe 'Unit Test of capitalize_decorator' do
+  before(:each) do
+    @capitalize_decorator = CapitalizeDecorator.new
+  end
+
```
Comment (tresorsawasawa) [inline spec/capitalize_decorator_spec.rb]: - [x] As required for the project to have unit tests for all the `methods` in all the `classes`, Kindly ensure that you add tests on the `correct_name` method in the `CapitalizeDecorator` class. This should test if the name is capitalized

File: spec/person_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  it 'should return false if not of_age and parent_permission' do
+    @person = Person.new(15, 'John Doe', parent_permission: false)
+    expect(@person.can_use_services?).to eql(false)
+  end
+
+  it 'check the be_instance_of person' do
+    expect(@person).to be_instance_of(Person)
+  end
+end
```
Comment (tresorsawasawa) [inline spec/person_spec.rb]: - [x] As required for the project to have unit tests for all the methods in all the classes, Kindly ensure that you add tests on the `of_age?`, `correct_name`, and `add_rental` methods in the `Person` class

File: spec/student_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    expect(@student.can_use_services?).to eql(false)
+  end
+
+  it 'should return the play_hooky of the student' do
+    expect(@student.play_hooky).to eql('¯\(ツ)/¯')
+  end
+
+  it 'check the be_instance_of student' do
+    expect(@student).to be_instance_of(Student)
+  end
```
Comment (tresorsawasawa) [inline spec/student_spec.rb:40]: - [x] As required for the project to have unit tests for all the methods in all the classes, Kindly ensure that you add tests on the `add_to_classroom` method in the `Student` class

File: spec/trimmer_decorator_spec.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+require_relative '../trimmer_decorator'
+
+describe 'Unit Test of trimmer_decorator' do
+  before(:each) do
+    @trimmer_decorator = TrimmerDecorator.new
+  end
+
```
Comment (tresorsawasawa) [inline spec/trimmer_decorator_spec.rb]: - [x] As required for the project to have unit tests for all the methods in all the classes, Kindly ensure that you add tests on the `correct_name` method in the `TrimmerDecorator` class which should test the `name` length characters

File: README.md [hunk 6]
```diff
@@ -95,6 +96,14 @@ To run the project, execute the following command:
...
+
+To run the tests, execute the following command:
+
+```sh
+  $ rspec
+```
+
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
 <!-- AUTHORS -->
```
Comment (tresorsawasawa) [inline README.md:109]: - [x] Kindly ensure that all your project's contributors are present in your `# Authors` section of your README to make your documentation more professional

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @yuvenalmash and @Hassaanjbaig-code 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- All tests are passing :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/trimmer_decorator_spec.rb [hunk 7]
```diff
@@ -8,4 +10,10 @@
   it 'should return NotImplementedError' do
     expect { @trimmer_decorator.correct_name }.to raise_error(NotImplementedError)
   end
+
+  it 'should return the trim name' do
+  person = Person.new(25, 'John Doe')
```
Comment (pinkmoon25) [inline spec/trimmer_decorator_spec.rb]: - [x] [OPTIONAL] In order to write solid test cases, you should provide a better example name here which actually is more than 11 characters then you can check that trimmer decorator works as expected.

**Review by pinkmoon25 — APPROVED** [review]
Hi @yuvenalmash , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Rubocop Checks :heavy_check_mark: 
gitflow :heavy_check_mark: 
Good Readme :heavy_check_mark: 
Unit tests :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**